shell/source/ArcsLib.js

```diff
@@ -30,15 +32,18 @@ import {KeyManager} from '../../runtime/ts-build/keymgmt/manager.js';
 //Tracing.enable();
 
 const Arcs = {
-  version: '0.3',
+  version: '0.4',
```

Comment:
I thought we were already on 0.4, maybe this is from an older branch? Also the Firebase changes below and above, are those ok, should they be part of this PR?

Reply:
This version number isn't used anywhere and it had just become stale. I noticed and updated it. Technically not necessary for this PR but I hate to lose little changes like this, so I tend to leave them in.

Removing shell's access to Firebase in favor of Storage abstractions is part of this effort. Firebase marshaling in runtime has changed recently and it needed some cleanup here. E.g. shell's `firebase.js` is moved to `shell/lib` in this PR. There's more things to be done, I did try to keep it to a minimum. :)